### PR TITLE
[Federation]  Query validation

### LINKED ISSUE #522: Federated schemas should not require at least one query to be implemented
In a Federated environment, the Gateway instantiates the Query type by default. This means that an implementing services should _not_ be required to implement or extend a query. 

# Ideal Scenario
This is an example scenario of what is valid in Node and Java implementations. For example, it should be valid to expose a service that exposes no root queries, but only the federated query fields, like below.

Produced Query type:

**Example**: This is what the schemas would look like for two federated services:
## Product Service
product/schema.gql
```gql
extend type Query {
   products: [Product]
}

type Product {
  id: ID!
  name: String
  reviews: [ProductReview]
}

extend type ProductReview @key(fields: "id") {
  id: ID! @external
}
```
**Output**:

```
products: [Product]
_entities(representations: [_Any]): [_Entity]
_service: _Service
```

## Review Service
review/schema.gql
```gql
# Notice how we don't have to extend the Query type
type ProductReview @key(fields: "id") {
   id: ID!
   comment: String!
}
```

**Output**:
This should be valid.
```
_entities(representations: [_Any]): [_Entity]
_service: _Service
```

# Breaking Scenario
When attempting to implement the `ProductReview` service (see example above) without extending the Query type, Ariadne will fail to [generate a federated schema](https://github.com/mirumee/ariadne/blob/master/ariadne/contrib/federation/schema.py#L57). This is because `make_executable_schema` attempts to generate a federated schema by [extending a Query type](https://github.com/mirumee/ariadne/blob/master/ariadne/contrib/federation/schema.py#L24) with the assumption that a Query type has been defined, which technically it isn't.

### PR DESCRIPTION
**Pull request by kuchichan**
Fixes #522, initally tested with: https://github.com/kuchichan/ariadne-federation-multiple-keys (Query removed from product service)

Todo:
- [x] Write real unit test
- [x] Remove a `repeatable` in key directive (It's a matter of different PR)

### CODE REVIEW

File: ariadne/contrib/federation/schema.py [hunk 1]
```diff
@@ -52,8 +64,12 @@ def make_federated_schema(
     # Remove custom schema directives (to avoid apollo-gateway crashes).
     # NOTE: This does NOT interfere with ariadne's directives support.
     sdl = purge_schema_directives(type_defs)
+    extend_token = "extend " if has_query_type(sdl) else ""
```
Comment (rafalp) [inline ariadne/contrib/federation/schema.py:67]: I'm thinking if this wouldn't be readable if we inserted `extend type` or `type` into query template.

File: tests/federation/test_schema.py [hunk 2]
```diff
@@ -720,3 +720,28 @@ def test_federated_schema_query_service_ignore_custom_directives():
...
+        name: String
+        price: Int
+        weight: Int
+    }
+    """
+
+    schema = make_federated_schema(type_defs)
+    result = graphql_sync(
+        schema,
+        """query GetServiceDetails {
```
Comment (rafalp) [inline tests/federation/test_schema.py:738]:
```suggestion
        """
        query GetServiceDetails {
```

File: ariadne/contrib/federation/schema.py (hunk 1 above)
Comment (kuchichan) [reply ariadne/contrib/federation/schema.py:67]: Agreed 👌

Comment (kuchichan) [reply ariadne/contrib/federation/schema.py]: Done.

File: tests/federation/test_schema.py (hunk 2 above)
Comment (kuchichan) [reply tests/federation/test_schema.py]: Done.

File: ariadne/contrib/federation/schema.py [hunk 3]
```diff
@@ -17,18 +18,18 @@
...
+    {type_token} Query {{
         _service: _Service!
-    }
+    }}
 
     directive @external on FIELD_DEFINITION
     directive @requires(fields: String!) on FIELD_DEFINITION
     directive @provides(fields: String!) on FIELD_DEFINITION
-    directive @key(fields: String!) on OBJECT | INTERFACE
+    directive @key(fields: String!)  on OBJECT | INTERFACE
```
Comment (rafalp) [inline ariadne/contrib/federation/schema.py:32]:
```suggestion
    directive @key(fields: String!) on OBJECT | INTERFACE
```

File: tests/federation/test_schema.py [hunk 4]
```diff
@@ -810,3 +810,29 @@ def test_federated_schema_query_service_ignore_custom_directives():
...
+        price: Int
+        weight: Int
+    }
+    """
+
+    schema = make_federated_schema(type_defs)
+    result = graphql_sync(
+        schema,
+        """
+        query GetServiceDetails {
```
Comment (rafalp) [inline tests/federation/test_schema.py:829]:
```suggestion
            query GetServiceDetails {
```

**Comment by rafalp** [comment]
@kuchichan LGTM! Could you also add line about this to changelog? Eg.:

"Made `Query` type optional in federated schemas."

**Review by salwator — APPROVED** [review]

**Comment by salwator** [comment]
@rafalp maybe we could merge this PR?

**Review by rafalp — APPROVED** [review]

**Comment by rafalp** [comment]
@salwator yup, lets merge!

**Pull request merged**